internal/dag/cache.go

```diff
@@ -532,6 +532,10 @@ func (kc *KubernetesCache) secretTriggersRebuild(secret *v1.Secret) bool {
 				return true
 			}
 		}
+		cv := tls.ClientValidation
+		if cv != nil && proxy.Namespace == secret.Namespace && cv.CertificateRevocationList == secret.Name {
```

Comment:
also add the delegation check like here? https://github.com/projectcontour/contour/blob/d16a115d1c49583309707f8aebb3b48b9f54e71a/internal/dag/cache.go#L525-L534

Reply:
I've now fixed this.

I suggest some refactoring for the comparison logic for delegations since in my opinion the old one was bit difficult to understand. 

How I _think_ the old logic worked: 

It built map `delegations` where it "projects" all delegated secrets like those secrets would have been in the target namespace ([link](https://github.com/projectcontour/contour/blob/d0f1171bd31f8aebf1fb80a6991138f396470d37/internal/dag/cache.go#L517-L523). For example, delegation of secret `foo` stored in `ns1` and delegated for namespace `targetns` would yield map entry for secret `targetns/foo`. I think of this as "projecting" the secret `foo` into target namespace.

Consider example where there were two secrets both called `foo`, one in `ns1` and another in `ns2`. Only `foo` in `ns1` was delegated. Now imagine that user refers to `ns2/foo` (which was not delegated).  The `delegations` list has the "projected" secret `targetns/foo` so the code accepts this and proceeds to compare secret reference `ns2/foo` - even if `ns2` was not delegated ([link](https://github.com/projectcontour/contour/blob/d0f1171bd31f8aebf1fb80a6991138f396470d37/internal/dag/cache.go#L565-L569)).  This is not that important; it would just trigger extra DAG rebuild when it does not need to.

In the refactored version, I keep the delegation checks, which complicates the logic but avoids the extra DAG rebuilds.